src/shared/reducers/status.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Redux Reducer for application-wide API/network status handling.
+ *
+ * Description:
+ *   Implements state reducers for application-wide API/network status handling.
+ */
+import { handleActions } from 'redux-actions';
+import actions from 'actions/status';
+
+/**
+ * Creates an Errors reducer with the specified initial state.
+ * @param {Array} initialState Initial state.
+ * @return submission reducer.
+ */
+function create(initialState) {
+  const a = actions.status;
+
+  return handleActions({
+    [a.clearAllErrorsStatus]: () => ({ networkErrors: {}, ApiErrors: {} }),
+    // setNetworkErrorsStatus only keep latest error message
+    [a.setNetworkErrorsStatus]: (state, { payload: { url, message } }) =>
+      ({ ...state, networkErrors: { [url]: message } }),
+    [a.clearNetworkErrorsStatus]: state => ({ ...state, networkErrors: {} }),
+    [a.setApiErrorsStatus]: (state, { payload: { url, message } }) =>
+      ({ ...state, ApiErrors: { ...state.ApiErrors, [url]: message } }),
+    [a.clearApiErrorsStatus]: (state, { payload }) => {
+      if (payload in state.ApiErrors) {
+        const newApiErrors = { ...state.ApiErrors };
+        delete newApiErrors[payload];
+        return { ...state, ApiErrors: newApiErrors };
+      }
+      return state;
+    },
+
+  }, initialState || { networkErrors: {}, ApiErrors: {} });
```

Comment:
the structure of related state part (after you merge it with the errors state) should be something like this:
```
{
  ...
  errorIcons: {
    errorIconId: [{
      url,
      msg,
    }]
  }
}
```
I mean, the point is that you don't specify the actual types of the icon, keeping them dynamic. In fact, we can even permit to add unknow error types to show there (we'll just have to check that we don't have an icon for them, but we can add a generic error icon).